keras_nlp/evaluations/glue/glue.py

```diff
@@ -0,0 +1,287 @@
+    # where features is a tuple of all input sentences.
+    if task_name in ("cola", "sst2"):
+        feature_names = ("sentence",)
+    elif task_name in ("mrpc", "stsb", "rte", "wnli"):
+        feature_names = ("sentence1", "sentence2")
+    elif task_name in ("mnli", "mnli_matched", "mnli_mismatched"):
+        feature_names = ("premise", "hypothesis")
+    elif task_name in "qnli":
+        feature_names = ("question", "sentence")
+    elif task_name in "qqp":
+        feature_names = ("question1", "question2")
+    else:
+        raise ValueError(f"Unknown task_name {task_name}.")
+
+    test_suffix = ""
+    if task_name in ("mnli", "mnli_matched"):
+        # For "mnli", just run default to "mnli_matched".
+        task_name = "mnli"
+        test_suffix = "_matched"
+    elif task_name in ("mnli_mismatched",):
+        task_name = "mnli"
+        test_suffix = "_mismatched"
+
+    def split_features(x):
+        # GLUE comes with dictonary data, we convert it to a uniform format
+        # (features, label), where features is a tuple consisting of all
+        # features.
+        features = tuple([x[name] for name in feature_names])
+        label = x["label"]
+        return (features, label)
+
+    train_ds, test_ds, validation_ds = tfds.load(
+        f"glue/{task_name}",
+        split=["train", "test" + test_suffix, "validation" + test_suffix],
+    )
+    train_ds = train_ds.map(split_features, num_parallel_calls=tf.data.AUTOTUNE)
+    test_ds = test_ds.map(split_features, num_parallel_calls=tf.data.AUTOTUNE)
+    validation_ds = validation_ds.map(
+        split_features, num_parallel_calls=tf.data.AUTOTUNE
+    )
+    return train_ds, test_ds, validation_ds
+
+
+def preprocess_data(preprocess_fn, dataset):
+    # Run `proprocess_fn` on input dataset then batch & prefetch.
+    return (
+        dataset.map(preprocess_fn)
+        .batch(FLAGS.batch_size)
+        .prefetch(tf.data.AUTOTUNE)
+    )
+
+
+def generate_submission_files(finetuning_model, test_ds):
+    # Generate GLUE leaderboard submission files.
+    filenames = {
+        "cola": "CoLA.tsv",
+        "sst2": "SST-2.tsv",
+        "mrpc": "MRPC.tsv",
+        "qqp": "QQP.tsv",
+        "stsb": "STS-B.tsv",
+        "mnli_matched": "MNLI-m.tsv",
+        "mnli_mismatched": "MNLI-mm.tsv",
+        "qnli": "QNLI.tsv",
+        "rte": "RTE.tsv",
+        "wnli": "WNLI.tsv",
+    }
+
+    labelnames = {
+        "mnli_matched": ["entailment", "neutral", "contradiction"],
+        "mnli_mismatched": ["entailment", "neutral", "contradiction"],
+        "qnli": ["entailment", "not_entailment"],
+        "rte": ["entailment", "not_entailment"],
+    }
+    if not os.path.exists(FLAGS.submission_file_path):
+        os.makedirs(FLAGS.submission_file_path)
+    filename = FLAGS.submission_file_path + "/" + filenames[FLAGS.task_name]
+
+    # This format is used for distribution strategy compatibility.
+    def eval_step(iterator):
```

Comment:
Can we not just use `predict` here? Shouldn't predict be distribution strategy compatible?

We could just call predict on the entire test set, than map over with an argmax in pure numpy on the output. (it is not important to distribute the argmax, running that cpu will be fine)